internal/injector/aspect/advice/order.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package advice
+
+import (
+	"sort"
+)
+
+const (
+	// DefaultNamespace is used when no namespace is specified
+	DefaultNamespace = "default"
```

Comment:
So the namespace participates in ordering -- in the sense that advice is ordered by lexicographic order of `namespace`, then `order`...

And now `default` occupies an arbitrary position that might be...surprising? Using an empty string would, IIUC, make non-namespaced advice consistently apply before namespaced advice... Or maybe we want them to consistently apply after all namespaced advice?

Anyhow -- I think the choice of this value is not neutral and folks _may_ occasionally need to be able to decide if they're before or after this (or do they?). Furthermore, changing this value in the future is a potentially breaking change, so we have to ask ourselves the question _now_.